### ISSUE TITLE
PLATFORM: Implement Google, GitLab and GitHub

### ISSUE THREAD

**Issue by kriptonian1** [keyshade-xyz/keyshade#307]
## Description

Implement Google, GitLab, and GitHub to the `/auth` so that user can use OAuth to log in to our application.

## Scope of Work

1. You need to implement Google, GitLab, and GitHub end points to the platform's [`/auth`](https://github.com/keyshade-xyz/keyshade/blob/b272df3210e7338be0722d6711cf110998500cb7/apps/platform/src/app/auth/page.tsx). 
2. After the authentication is successful, it will automatically be redirected to the [`/auth/account-details`](https://github.com/keyshade-xyz/keyshade/blob/b272df3210e7338be0722d6711cf110998500cb7/apps/platform/src/app/auth/account-details/page.tsx) page from the backend.

## URL endpoints with Postman Links

- Google Auth -> [`{{BASE_URL}}/api/auth/google`](url)
- GitHub Auth -> [`{{BASE_URL}}/api/auth/github`](url)
- GitLab Auth -> [`{{BASE_URL}}/api/auth/gitlab`](url)

## Figma link
- Google Auth flow [link](https://www.figma.com/design/Du3qt5NWbbnfMgtU5CQYCU/Keyshade-UI?node-id=862-468&node-type=canvas&t=MepyHS8rqzm3B3fp-0)
- GitHub Auth flow [link](https://www.figma.com/design/Du3qt5NWbbnfMgtU5CQYCU/Keyshade-UI?node-id=862-469&node-type=canvas&t=MepyHS8rqzm3B3fp-0)
- GitLab Auth flow [link](https://www.figma.com/design/Du3qt5NWbbnfMgtU5CQYCU/Keyshade-UI?node-id=862-470&node-type=canvas&t=MepyHS8rqzm3B3fp-0)

**Comment by harshdeshmukh21** [comment]
I would like to work on this issue. I have worked with OAuth before and I feel I can do this task. I have registered my team via Foss Hack.

**Comment by rajdip-b** [comment]
Hey @harshdeshmukh21! The implementation of OAuth is done in the backend. We would just need you to tie up the logic with the frontend. But yes, go ahead!

**Comment by harshdeshmukh21** [comment]
I am able to run the api and web app locally. Since my issue is platform related, how can I run platform locally? @rajdip-b

**Comment by rajdip-b** [comment]
It is exactly similar to how you run web. We do need to work on the docs :)

**Comment by harshdeshmukh21** [comment]
Can I get the client ids and secrets for Google, GitHub and GitLab or do I have to create clients for testing?

**Comment by harshdeshmukh21** [comment]
Also is there any message after running all the apis sucessfully, because I feel there is some kind of error and I am not able to access all the routes. Only the /auth route is accessible.

**Comment by rajdip-b** [comment]
> Can I get the client ids and secrets for Google, GitHub and GitLab or do I have to create clients for testing?

We would have loved to, but then we are also using it in our prod :(

> Also is there any message after running all the apis sucessfully? Because I feel there is some kind of error and I am not able to access all the routes. Only the /auth route is accessible.

Most of the errors are originated from env stuff. You can give this a read: https://docs.keyshade.xyz. Additionally, you can post screenshots in here.

**Comment by kriptonian1** [comment]
Hey @harshdeshmukh21, To run everything locally, you need to go through the docs and st the thing up, and to run the web you can follow this comment https://github.com/keyshade-xyz/keyshade/issues/244#issuecomment-2249252029, here it is mentioned how to run the platform.

**Comment by harshdeshmukh21** [comment]
Can I get the SMTP credentials for running the API?

**Comment by harshdeshmukh21** [comment]
Even after setting up SMTP with google, I am not able to run the api completely. Apparently, it is stuck on this. 
<img width="796" alt="Screenshot 2024-07-27 at 5 25 33 PM" src="https://github.com/user-attachments/assets/fb6293ae-f3d7-43dc-b1c1-ccbcaa11703f">

**Comment by rajdip-b** [comment]
The `FROM_EMAIL` must be in this format:
![image](https://github.com/user-attachments/assets/b0d6ec71-acc4-4a4b-b217-fc8148e9e4cd)

**Comment by kriptonian1** [comment]
@harshdeshmukh21 the email must be in the format of `FROM_EMAIL="harsh <harsh.deshmukh21@email.com>"`

**Comment by kriptonian1** [comment]
@harshdeshmukh21 are you still working on this issue?

**Comment by harshdeshmukh21** [comment]
> @harshdeshmukh21 are you still working on this issue?

Nope. I tried a lot but wasn't able to do it. Thanks for assigning me the issue. You may reassaign it to someone else.

**Comment by dev-palwar** [comment]
Hey @kriptonian1  Would it be okay if I take this on? I've done it before

**Comment by rajdip-b** [comment]
Yeah you can go ahead @dev-palwar!

**Comment by rajdip-b** [comment]
Closed as complete
